dev-docs/README.md

```diff
@@ -0,0 +1,40 @@
+# Developer Documentation
```

Comment:
It's not clear to me why this is a separate document from CONTRIBUTING.md and not linked from it.